### PR TITLE
Add hack language support

### PR DESCRIPTION
**Pull request by cotsog**
Please notify @fredemmott [in this ticket](https://secure.helpscout.net/conversation/508977913/65718/) when it's released.

### CODE REVIEW

File: lib/travis/build/script/php.rb [hunk 1]
```diff
@@ -100,7 +100,11 @@ def cache_slug
         private
 
         def version
-          config[:php].to_s
+          if config[:hhvm]
+            "hhvm-#{config[:hhvm].to_s}"
```
Comment (fredemmott) [inline lib/travis/build/script/php.rb]: What would be the equivalent of:

```
php:
 - hhvm
 - hhvm-nightly
```

i.e. how would I specify that one of the things that I want to run is the latest version of HHVM?

Comment (cotsog) [reply lib/travis/build/script/php.rb]: @fredemmott Thanks for chiming in!

How about if we supported `latest` in that case e.g. the equivalent of your snippet with `language: hack` would be

```yaml
hack:
  - latest
  - nightly
```

What do you think?

Note: the current state of the PR doesn't support this yet.

**Comment by fredemmott** [comment]
*bump* the next release of HHVM will not support PHP.

**Comment by BanzaiMan** [comment]
@fredemmott Thanks for the updates. We appreciate it. Do you know when the next HHVM is released?

**Comment by fredemmott** [comment]
Due 11th of February

**Comment by fredemmott** [comment]
*bump*

**Comment by fredemmott** [comment]
HHVM 4.0 is released, an d can not execute Composer.

For now, we're going to have to revert to docker-based testing.

**Comment by joshk** [comment]
@BanzaiMan @cotsog could we help get this merged and deployed?

Should this be separated from the PHP lang defaults? eg. its own class? `Script::Hack`

**Comment by fredemmott** [comment]
> Should this be separated from the PHP lang defaults? eg. its own class? `Script::Hack`

Yep - HHVM 4 can not execute Composer, or most other PHP code. In 4.1, I'm likely disabling support for `<?php` as it basically indicates the file won't work anyway

**Comment by BanzaiMan** [comment]
I'm closing this PR, because Hack looks to be different enough from PHP, and we will need a separate class to deal with the differences.

**Pull request closed without merging**